File: readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string s);
int count_words(string s);
int count_sentences(string s);
int coleman_index(int letras, int palabras, int oraciones);

int main(void)
{
    char *s = get_string("Prompt: ");
    
    int a = count_letters(s);
    int b = count_words(s);
    int c = count_sentences(s);
    int n = coleman_index(a, b, c);
    
    if (n < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (n >= 1 && n <= 16)
    {
        printf("Grade %i\n", n);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

int coleman_index(int letras, int palabras, int oraciones)
{
    float L = (letras * 100) / (float) palabras;
    float S = (oraciones * 100) / (float) palabras;
    return round(0.0588 * L - 0.296 * S - 15.8);
}

// Function for counting letters
int count_letters(char *parrafo)
{
    int count = 0;
    for (int i = 0, n = strlen(parrafo); i < n; i++)
    {
        if ((*(parrafo + i) >= 'A' && *(parrafo + i) <= 'Z') || (*(parrafo + i) >= 'a' && *(parrafo + i) <= 'z'))
        {
            count++;
        }
        else
        {
            
        }
    }
    return count;
}

// Function for counting words
int count_words(string parrafo)
{
    int count = 0;
    for (int i = 0, n = strlen(parrafo); i < n; i++)
    {
        if (*(parrafo + i) == ' ')
        {
            count++;
        }
    }
    return count + 1;
}

// Function for counting sentences
int count_sentences(string parrafo)
{
    int count = 0;
    for (int i = 0, n = strlen(parrafo); i < n; i++)
    {
        if (*(parrafo + i) == '!' || *(parrafo + i) == '.' || *(parrafo + i) == '?')
        {
            count++;
        }
    }
    return count;
}

File: readability.py
from cs50 import get_string


def main():
    s = get_string("Prompt: ")

    letters = count_letters(s)
    words = count_words(s)
    sentences = count_sentences(s)
    n = coleman_index(letters, words, sentences)

    if n < 1:
        print("Before Grade 1")
    elif n >= 1 and n <= 16:
        print(f"Grade {n}")
    else:
        print("Grade 16+")


def coleman_index(a, b, c):
    L = (a * 100) / b
    S = (c * 100) / b
    return round(0.0588 * L - 0.296 * S - 15.8)


def count_letters(parrafo):
    countl = 0
    for i in (parrafo):
        if i.isalpha():
            countl += 1
    return countl


def count_words(parraf):
    countw = 0
    for i in (parraf):
        if i == " ":
            countw += 1
    return countw + 1


def count_sentences(prompt):
    counts = 0
    for i in (prompt):
        if i == "!" or i == "." or i == "?":
            counts += 1
    return counts


main()